server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -83,7 +94,7 @@
             30 * 24 * 60 * 60 * 1000L, "M", 1, 3);
         roundings[5] = new RoundingInfo(createRounding(Rounding.DateTimeUnit.YEAR_OF_CENTURY, timeZone),
             365 * 24 * 60 * 60 * 1000L, "y", 1, 5, 10, 20, 50, 100);
-        return roundings;
+        return Arrays.copyOfRange(roundings, indexToSliceFrom, roundings.length);
```

Comment:
The way we are using the `minimumInterval` here relies on the `roundings` and the `allowedIntervals` being in sync (the contents and positions being lined up) but there is nothing that enforces that. I wonder if we should approach this more like the [`DateHistogramAggregationBuilder.DATE_FIELD_UNITS`](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java#L77-L93) and have a map of input String to `Rounding.DateTimeUnit`? The `AutoDateHistogramAggregationBuilder` would then hold a `Rounding.DateTimeUnit` for `minimumInterval` (the parser would do the conversion from the String using the map) and we could resolve the range of the array to keep here be finding the `minimumInterval` directly in the array?

Reply:
++ I like this approach.

Reply:
soooooo, @colings86 and I discussed this and it turns out we felt it's a bit misleading to type the minimum interval (and allowed intervals) to a `Rounding.DateTimeUnit`, because we only support a subset of these, so it's possible to pass a valid `Rounding.DateTimeUnit` and get an exception, which could be surprising to a HLRC user. So, I've left it as a string, but made two changes:

1) refactor to use a map to remove duplication of string literals
2) iterate through roundings instead of allowed intervals to determine slice point. this should address the maintainability concern that @colings86 expressed.